cpp/new_journal_alert.cc

```diff
@@ -418,12 +420,56 @@ void LoadBundleControlNumbers(const IniFile &bundles_config, const std::string &
 }
 
 
+std::function<std::string(std::string)> SingleQuote = [](std::string to_quote) { return "'" + to_quote + "'"; };
+
+void LoadBundleMaxLastModificationTimes(DbConnection * const db_connection, const std::string &bundle_name,
+                                        std::vector<std::string> bundle_control_numbers,
+                                        std::map<std::string, std::string> * const bundle_journals_max_last_modification_times)
+{
+    bundle_journals_max_last_modification_times->clear();
+    std::vector<std::string> quoted_bundle_control_numbers;
+    std::transform(bundle_control_numbers.begin(), bundle_control_numbers.end(), std::back_inserter(quoted_bundle_control_numbers), SingleQuote);
+    db_connection->queryOrDie("SELECT journal_control_number, max_last_modification_time FROM ixtheo_journal_bundles WHERE bundle_name='"
+                               + bundle_name + "' AND journal_control_number IN (" +
+                               StringUtil::Join(quoted_bundle_control_numbers, ',') +
+                               ')');
+    DbResultSet result_set(db_connection->getLastResultSet());
+    while (const auto &row = result_set.getNextRow())
+        bundle_journals_max_last_modification_times->emplace( std::make_pair(row["journal_control_number"], row["max_last_modification_time"]));
+}
+
+
+bool IsBundle(const std::string &serial_control_number) {
+    if (serial_control_number.empty())
+        return false;
+    return not std::isdigit(serial_control_number[0]);
+}
+
+
+bool MinModificationTimeComparator(const std::pair<std::string, std::string> elem1, std::pair<std::string, std::string> elem2) {
```

Comment:
A better name please like ...LessThan..>?  Also, the second argument should be const, too.